src/main/java/fi/aalto/cs/apluscourses/model/Exercise.java

```diff
@@ -123,7 +123,12 @@ public int getMaxSubmissions() {
   }
 
   public boolean isCompleted() {
-    return userPoints == maxPoints && submissionResults.size() > 0;
+    // Optional assignments are never completed, since they can be filtered separately
```

Comment:
`...and we can't tell from the points whether the submission was correct or not`